### PR TITLE
feat(execution): allow flowable parallel tasks to fail-fast they chil…

### PR DESCRIPTION
**Pull request by loicmathieu**
…d tasks

Parallel and Dag can have one failing tasks and other still processing concurrenctly which would lead to those tasks being run un-necessary.

You can now configure `onChildFailure`  on both so when one child task fail, the worker can be asked to interrupt any still running child tasks in FAILED or CANCELLED.

https://github.com/kestra-io/kestra/issues/15365

## Breaking change

The default is `FAILED` so now Dag and Parallel will fail concurrently running tasks when one task fail.
Use `CONTINUE` to keep the previous behavior.

### CODE REVIEW

**Comment by loicmathieu** [comment]
@fhussonnois FYI this PR adds a new capacity to the Worker to interrupt individual tasks (and not the whole execution) in a target state. I plan to allow doing that for any task run from the API later in 2.1 as described some time ago in https://github.com/kestra-io/kestra/issues/17569

File: executor/src/main/java/io/kestra/executor/ExecutorService.java [hunk 1]
```diff
@@ -841,6 +852,59 @@ else if (executor.getFlow().getRetry() != null) {
         return executor;
     }
 
+    /**
+     * Interrupts every still-running task run in the task subtree so they don't keep executing on an already-failed branch.
+     * Send an {@link ExecutionKilledTaskRuns} event to the Worker to interrupt the non-terminated task runs.
+     */
+    private void interruptOnChildFailure(ExecutorContext executor, OnChildFailureInterface onChildFailure, TaskRun parentTaskRun, RunContext runContext) throws InternalException {
```
Comment (AcevedoR) [inline executor/src/main/java/io/kestra/executor/ExecutorService.java:859]: this method might be a good candidate for a unit test suite that covers edge cases

**Review by AcevedoR — APPROVED** [review]

**Comment by loicmathieu** [comment]
@Ben8t it contains a new (last minute) small breaking change
